Bourreau/app/models/bourreau_worker.rb

```diff
@@ -276,6 +279,13 @@ def process_task(task) # when entering this methods task is a partial object, wi
 
       worker_log.debug "Updated #{task.bname_tid} to state #{new_status}"
 
+      # Mechanism for tasks to submit other tasks: tasks that were not in a status 
+      # in COMPLETED_STATUS or FAILED_STATUS at the previous iteration may submit new
+      # tasks provided that they have the can_submit_new_tasks
+      # property.
+      handle_tasks_submitted_by(task) if task.class.properties[:can_submit_new_tasks] &&
+                                         !(CbrainTask::COMPLETED_STATUS.include?(task.status) || CbrainTask::FAILED_STATUS.include?(task.status))
```

Comment:
Not sure about the two sets of states here... can we do a check in the opposite direction (e.g. if state is among a precise set of states?)